src/main/java/seedu/address/model/consultation/ConsultationList.java

```diff
@@ -54,6 +56,39 @@ public void remove(Consultation toRemove) {
         }
     }
 
+    /**
+     * Removes the {@code student} from all consultations in the consultation list.
+     */
+    public void removePerson(Person student) {
+        requireNonNull(student);
+        List<Consultation> newConsultations = new ArrayList<>();
+
+        for (Consultation consultation : internalList) {
+            Consultation updatedConsultation = consultation.removeStudent(student);
+            if (!updatedConsultation.isEmpty()) {
+                newConsultations.add(updatedConsultation);
+            }
+        }
+        internalList.setAll(newConsultations);
+    }
+
+    /**
+     * Replaces the student {@code target} in the list with {@code editedStudent}.
+     */
+    public void setStudent(Person target, Person editedStudent) {
+        requireAllNonNull(target, editedStudent);
+        List<Consultation> newConsultations = new ArrayList<>();
+
+        for (Consultation consultation : internalList) {
+            if (consultation.contains(target)) {
+                consultation.removeStudent(target);
```

Comment:
perhaps can abstract the 2 commands here into a replace command in Consultation

Reply:
Sounds like a good idea!